templates/rule-header.erb

```diff
@@ -36,7 +36,7 @@
 <% if example['test_values'] -%>
 						<test_values>
 <% Hash[example['test_values']].sort.each do |k,v| -%>
-							<test_value name='<%=k%>'><%=scope.call_function('htmlentities', [v])%></test_value>
+							<test_value name='<%=k%>' type='<%= case v.class when Float then 'double' else v.class.name.downcase end %>'><%=scope.call_function('htmlentities', [v])%></test_value>
```

Comment:
This will work with ruby `String` ("string"), `Integer` ("integer") and `Float` ("double"), but not for the data types that are supposed to be interpreted as "datetime" or "json": https://github.com/bazsi/syslog-ng/blob/193b8819c6d00ae094df05ee8e360468ece59962/doc/xsd/patterndb-6.xsd#L193-L196